python-sdk/src/astro/sql/operators/data_validations/ColumnCheckOperator.py

```diff
@@ -0,0 +1,242 @@
+            dataset_qualified_name = db.get_table_qualified_name(table=self.dataset)
+            dataset_conn_id = dataset.conn_id
+
+        super().__init__(
+            table=dataset_qualified_name,
+            column_mapping=self.column_mapping,
+            partition_clause=self.partition_clause,
+            conn_id=dataset_conn_id,
+            task_id=task_id if task_id is not None else get_unique_task_id("column_check"),
+        )
+
+    def execute(self, context: "Context"):
+        if isinstance(self.dataset, BaseTable):
+            return super().execute(context=context)
+        elif type(self.dataset) == pandas.DataFrame:
+            self.df = self.dataset
+        else:
+            raise ValueError("dataset can only be of type pandas.dataframe | Table object")
+
+        self.process_checks()
+
+    def get_check_result(self, check_name: str, column_name: str, df: pandas.DataFrame):
+        """
+        Get the check method results post validating the dataframe
+        """
+        if df is not None and column_name in df.columns:
+            column_checks = {
+                "null_check": self.col_null_check,
+                "distinct_check": self.col_distinct_check,
+                "unique_check": self.col_unique_check,
+                "min": self.col_min,
+                "max": self.col_max,
+            }
+            return column_checks[check_name](column_name=column_name, df=df)
+        elif df is None:
+            raise ValueError("Dataframe is None")
+        else:
+            raise ValueError(f"Dataframe is don't have column {column_name}")
+
+    def process_checks(self):
+        """
+        Process all the checks and print the result or raise an exception in the event of failed checks
+        """
+        failed_tests = []
+        passed_tests = []
+
+        # Iterating over columns
+        for column in self.column_mapping:
+            checks = self.column_mapping[column]
+
+            # Iterating over checks
+            for check in checks:
+                tolerance = self.column_mapping[column][check].get("tolerance")
+                result = self.get_check_result(check, column_name=column, df=self.df)
+                self.column_mapping[column][check]["result"] = result
+                self.column_mapping[column][check]["success"] = self._get_match(
+                    self.column_mapping[column][check], result, tolerance
+                )
+                failed_tests.extend(_get_failed_checks(self.column_mapping[column], column))
+                passed_tests.extend(_get_success_checks(self.column_mapping[column], column))
+
+        if len(failed_tests) > 0:
+            raise AirflowException(f"The following tests have failed:" f"\n{''.join(failed_tests)}")
+        if len(passed_tests) > 0:
+            print(f"The following tests have passed:" f"\n{''.join(passed_tests)}")
+
+    @staticmethod
+    def col_null_check(column_name: str, df: pandas.DataFrame) -> Optional[int]:
```

Comment:
```suggestion
    def col_null_check(column_name: str, df: pandas.DataFrame) -> int:
```
All the check functions return type should be non-Optional now right?